lighthouse-core/gather/gatherers/image-usage.js

```diff
@@ -0,0 +1,149 @@
+/**
+ * @license
+ * Copyright 2017 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+ /**
+  * @fileoverview Gathers all images used on the page with their src, size,
+  *   and attribute information. Executes script in the context of the page.
+  */
+'use strict';
+
+const Gatherer = require('./gatherer');
+
+/* global window, document, Image */
+
+/* istanbul ignore next */
+function collectImageElementInfo() {
+  function parseSrcsetUrls(srcset) {
+    if (!srcset) {
+      return [];
+    }
+
+    const entries = srcset.split(',');
+    const relativeUrls = entries.map(entry => entry.trim().split(' ')[0]);
+    return relativeUrls.map(url => {
+      const a = document.createElement('a');
```

Comment:
Nice trick to get full urls. Any value in using`new URL()` with the 2nd base param instead of creating dom? I guess what you have also works if there's  `<base>` tag on the page.